plugins/services/src/js/structs/Framework.js

```diff
@@ -97,13 +97,35 @@ module.exports = class Framework extends Application {
     // used_resources is actually allocated resources
     // the name can't be changed to keep backward compatibility.
     // This is why `getResources` returns `used_resources`
-    return (
-      this.get("used_resources") || {
-        cpus: 0,
-        mem: 0,
-        gpus: 0,
-        disk: 0
-      }
-    );
+    const allocatedFrameworkResources = this.get("used_resources") || {
+      cpus: 0,
+      mem: 0,
+      gpus: 0,
+      disk: 0
+    };
+
+    // TODO: Circular reference workaround DCOS_OSS-783
+    const MesosStateStore = require("#SRC/js/stores/MesosStateStore");
```

Comment:
🤔 Wonder whether we can read the scheduler resource form the spec using `this.getSpec().getResources();` instead of using the Mesos state store.

Reply:
We can explore the possibility

Reply:
IMHO: create JIRA and fix later 👍 (I'll approve once the JIRA is there)

Reply:
I think it makes sense exploring it on the spot. We have 8 SP planned and the performance win can be big.

Reply:
@nLight Do I get it right that you are then still working on this? Could we move the Jira into Progress then?

Reply:
There and back again 💪

Reply:
@orlandohohmeier you were right, it looks like we can just use the spec, at least numbers don't change 🤷‍♂️